test/test_linalg.py

```diff
@@ -410,6 +410,103 @@ def run_test_case(input, p, dim, keepdim):
                 for ord in ord_settings:
                     run_test_case(input, ord, dim, keepdim)
 
+    @skipCPUIfNoLapack
+    @skipCUDAIfNoMagma
+    @dtypes(torch.float32, torch.float64, torch.complex64, torch.complex128)
+    @dtypesIfCUDA(torch.float32, torch.float64)
+    @precisionOverride({torch.float32: 1e-3})
+    def test_cond(self, device, dtype):
+        def run_test_case(input, ord):
+            result = torch.linalg.cond(input, ord)
+            result_numpy = np.linalg.cond(input.cpu().numpy(), ord)
+            self.assertEqual(result, result_numpy, rtol=1e-2, atol=self.precision)
+
+            # test out= variant
+            out = torch.empty_like(result)
+            ans = torch.linalg.cond(input, ord, out=out)
+            self.assertEqual(ans, out)
+            self.assertEqual(ans, result)
+
+        norm_types = [1, -1, 2, -2, inf, -inf, 'fro', 'nuc', None]
+        input_sizes = [(3, 3), (2, 3, 3, 3)]
```

Comment:
Vary the last two dimensions between the examples and, if this supports non-square matrices, create a non-square case, too

Reply:
Non-square matrices are supported for {None, 2, -2} norms. I added it now to the tests.